File: JavaScript/Codility/BInaryGap.js
function solution(N) {
    var binaryN = N.toString(2);
    var maxGap = 0;
    var tmpGap = 0;

    for (var i=0; i<binaryN.length; i++){
        if (binaryN[i] == "1"){
            if (tmpGap > maxGap){
                maxGap = tmpGap;
            }
            tmpGap = 0;
        }
        else {
            tmpGap += 1;
        }
    }
    return maxGap;
}

File: Python/Codility/BinaryGap.py
def solution(N):
    data = str(format(N, 'b'))

    maxGap = 0
    tmpGap = 0
    for d in data:
        if d == '1':
            maxGap = max(maxGap, tmpGap)
            tmpGap = 0
        elif d == '0':
            tmpGap += 1

    return maxGap